tests/PromoCodesTest.php

```diff
@@ -0,0 +1,201 @@
+
+        $this->assertEquals(count($promoCodesBeforeCreate) + 1, count($promoCodesAfterCreate));
+
+        $this->assertEquals($promoCode, end($promoCodesAfterCreate));
+    }
+
+    public function testCannotListPromoCodesFromRemovedOrganization()
+    {
+        $this->client->createPromoCodeRequest($this->testMaintainerId, [
+            'code' => 'TEST-' . time(),
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc6months',
+        ]);
+
+        $listBeforeRemoveOrganization = $this->client->listPromoCodesRequest($this->testMaintainerId);
+
+        $this->client->deleteOrganization($this->organization['id']);
+
+        $listAfterRemoveOrganization = $this->client->listPromoCodesRequest($this->testMaintainerId);
+
+        $this->assertLessThan(count($listBeforeRemoveOrganization), count($listAfterRemoveOrganization));
+    }
+
+    public function testDifferentOrganizationMaintainerCannotCreatePromoCodes()
+    {
+        $testMaintainer = $this->client->getMaintainer($this->testMaintainerId);
+
+        $maintainerName = self::TESTS_MAINTAINER_PREFIX . " - test maintainer";
+        $newMaintainer = $this->client->createMaintainer([
+            'name' => $maintainerName,
+            'defaultConnectionMysqlId' => $testMaintainer['defaultConnectionMysqlId'],
+            'defaultConnectionRedshiftId' => $testMaintainer['defaultConnectionRedshiftId'],
+            'defaultConnectionSnowflakeId' => $testMaintainer['defaultConnectionSnowflakeId'],
+        ]);
+
+        $this->client->addUserToMaintainer($newMaintainer['id'], ['id' => $this->normalUser['id']]);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionCode(404);
+        $this->normalUserClient->createPromoCodeRequest($this->testMaintainerId, [
+            'code' => 'TEST-' . time(),
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc15DaysGuideMode',
+        ]);
+    }
+
+    public function testSuperAdminCanListAndCreatePromoCodes()
+    {
+        $promoCodesBeforeCreate = $this->client->listPromoCodesRequest($this->testMaintainerId);
+        $promoCode = $this->client->createPromoCodeRequest($this->testMaintainerId, [
+            'code' => 'TEST-' . time(),
+            'expirationDays' => rand(5, 20),
+            'organizationId' => $this->organization['id'],
+            'projectTemplateStringId' => 'poc15DaysGuideMode',
+        ]);
+        $promoCodesAfterCreate = $this->client->listPromoCodesRequest($this->testMaintainerId);
+
+        $this->assertEquals(count($promoCodesBeforeCreate) + 1, count($promoCodesAfterCreate));
+
+        $this->assertEquals($promoCode, end($promoCodesAfterCreate));
+    }
+
+    public function testCannotCreateDuplicatePromoCode()
```

Comment:
Pridal bych i test na hlasku, kterou to vraci.

Reply:
přidána :+1: